pkg/kubectl/cmd/util/factory.go

```diff
@@ -767,6 +770,27 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 				return false, fmt.Errorf("the object is not a pod or does not have a pod template")
 			}
 		},
+		// UpdateSelectorForObject update the selector for the provided object
+		UpdateSelectorForObject: func(obj runtime.Object, fn func(*map[string]string, *unversioned.LabelSelector) error) error {
```

Comment:
I'm ok with adding this, but its time to start building reflective tests around it.  Write a test that installs all the types, navigates the scheme to check them all for an `unversioned.LabelSelector` field, then makes sure that doesn't end up with unsupported error from the switch.